### PR TITLE
Dala 4834 search for specific qa reports

### PR DESCRIPTION
**Pull request by LaurinStrelow**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [x] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [x] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * Defines the restful dataland QAReports Metadata API
+ */
+@Suppress("kotlin:S6517")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface QaReportsMetadataApi {
+    /**
+     * A method to retrieve metadata of QA reports based on filters.
+     * @param uploaderUserIds If set, only metadata of reports are returned that relate to a dataset of an uploader
```
Comment (jenniferlechner) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:24]: Could fomulate more specific, e.g. ...tthat pertain to data uploaded by this user(s)

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     *  in uploaderUserIds.
+     * @param showOnlyActive If true, only metadata of reports are returned that are active
+     * @param qaStatus If set, only metadata of reports are returned that relate to a dataset with this qaStatus
+     * @param startDate If set, only metadata of reports are returned that are past the startDate
```
Comment (jenniferlechner) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:28]: to be more specific: .. that were uploaded after the startDate

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     * @param endDate If set, only metadata of reports are returned that are before the endDate
```
Comment (jenniferlechner) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:29]: to be more specific: .. that were uploaded before the endDate

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                .toEpochMilli()
+        }
+    }
+
+    /**
+     * Method to search all data and the connected meta information associated with a data set.
+     * @param uploaderUserIds set of user ids of the uploader
+     * @param showOnlyActive whether to show only active data
+     * @param qaStatus the qa status
+     * @param startDate start date of the search
```
Comment (jenniferlechner) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt:43]: shows reports uploaded after startdate

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+     * @param endDate end date of the search
```
Comment (jenniferlechner) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt:44]: shows reports uploaded before this date

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt [hunk 6]
```diff
@@ -24,9 +26,12 @@ interface MetaDataApi {
 
     /**
      * A method to search for meta info about data sets registered by Dataland
-     * @param companyId filters the requested meta info to a specific company.
-     * @param dataType filters the requested meta info to a specific data type.
+     * @param companyId if set, filters the requested meta info to a specific company.
+     * @param dataType if set, filters the requested meta info to a specific data type.
```
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt:30]: Yes, this is the framework and yes, it is super confusing. However, this name is used throughout the project, so I would prefer to stick with it for consistency.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt [hunk 7]
```diff
@@ -24,9 +26,12 @@ interface MetaDataApi {
...
      * @param showOnlyActive if set to true, only active datasets will be returned (e.g. no outdated ones)
```
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt:31]: Good point, I will change the wording.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt [hunk 8]
```diff
@@ -24,9 +26,12 @@ interface MetaDataApi {
...
+     * @param reportingPeriod if set, the method only returns meta info with this reporting period
+     * @param uploaderUserIds if set, the method only returns meta info for datasets uploader by this user ids.
```
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt:33]: I will change this.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt:31]: Changed.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/MetaDataApi.kt:33]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt (hunk 1 above)
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:24]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+ */
+@Suppress("kotlin:S6517")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface QaReportsMetadataApi {
+    /**
+     * A method to retrieve metadata of QA reports based on filters.
+     * @param uploaderUserIds If set, only metadata of reports are returned that relate to a dataset of an uploader
+     *  in uploaderUserIds.
+     * @param showOnlyActive If true, only metadata of reports are returned that are active
```
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:26]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt (hunk 2 above)
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:28]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt (hunk 3 above)
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/api/QaReportsMetadataApi.kt:29]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt (hunk 4 above)
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt:43]: Changed.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt (hunk 5 above)
Comment (LaurinStrelow) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportMetadataService.kt:44]: Changed.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt [hunk 10]
```diff
@@ -99,7 +99,7 @@ abstract class DataController<T>(
         val reportingPeriodInLog = reportingPeriod ?: "all reporting periods"
         logger.info(logMessageBuilder.getFrameworkDatasetsForCompanyMessage(dataType, companyId, reportingPeriodInLog))
         val metaInfos = dataMetaInformationManager.searchDataMetaInfo(
-            companyId, dataType, showOnlyActive, reportingPeriod,
+            companyId, dataType, showOnlyActive, reportingPeriod, null, null,
```
Comment (emanueldfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt:102]: Kurz drüber sprechen

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationEntity.kt [hunk 11]
```diff
@@ -72,15 +72,11 @@ data class DataMetaInformationEntity(
     }
 
     override fun toApiModel(viewingUser: DatalandAuthentication?): DataMetaInformation {
```
Comment (emanueldfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/entities/DataMetaInformationEntity.kt:74]: Drüber sprechen

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/vsme/VsmeDataController.kt [hunk 12]
```diff
@@ -105,7 +105,7 @@ class VsmeDataController(
             logMessageBuilder.getFrameworkDatasetsForCompanyMessage(dataType, companyId, reportingPeriodInLog),
         )
         val metaInfos = dataMetaInformationManager.searchDataMetaInfo(
-            companyId, dataType, showOnlyActive, reportingPeriod,
+            companyId, dataType, showOnlyActive, reportingPeriod, null, null,
```
Comment (emanueldfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/vsme/VsmeDataController.kt:108]: Kurz drüber sprechen

**Pull request merged**